packages/plugin-core/src/commands/ImportPod.ts

```diff
@@ -47,16 +47,26 @@ export class ImportPodCommand extends BaseCommand<
     const podChoice = inputs.podChoice;
     const podClass = podChoice.podClass;
     const podsDir = DendronWorkspace.instance().podsDir;
-    const maybeConfig = PodUtils.getConfig({ podsDir, podClass });
-    if (!maybeConfig) {
-      const configPath = PodUtils.genConfigFile({ podsDir, podClass });
-      await VSCodeUtils.openFileInEditor(Uri.file(configPath));
-      window.showInformationMessage(
-        "Looks like this is your first time running this pod. Please fill out the configuration and then run this command again. "
-      );
+    try {
+      const maybeConfig = PodUtils.getConfig({ podsDir, podClass });
+
+      if (!maybeConfig) {
+        const configPath = PodUtils.genConfigFile({ podsDir, podClass });
+        await VSCodeUtils.openFileInEditor(Uri.file(configPath));
+        window.showInformationMessage(
+          "Looks like this is your first time running this pod. Please fill out the configuration and then run this command again."
+        );
+        return;
+      }
+      return { podChoice, config: maybeConfig };
+    } catch (e) {
+      // The user's import configuration has YAML syntax errors:
+      if (e.name === "YAMLException")
+        window.showErrorMessage(
+          "The configuration is invalid YAML. Please fix and run this command again."
+        );
```

Comment:
if error is not a YAML error, we should let it propagate through. commands will log errors and display in UI if uncaught from the child command

Reply:
good catch, updated. let me know if there's better methodology on the re-throw; coming from Java/C# I'm finding exception handling to be a quite different in JS haha

Reply:
re-throw is the typical response. we have two patterns for error handling: throwing and catching in a base class or returning an error object